Mathlib/Algebra/CharP/Two.lean

```diff
@@ -37,16 +56,27 @@ variable [Ring R] [CharP R 2]
 
 @[scoped simp]
 theorem neg_eq (x : R) : -x = x := by
-  rw [neg_eq_iff_add_eq_zero, ← two_smul R x, two_eq_zero, zero_smul]
+  rw [neg_eq_iff_add_eq_zero, add_self_eq_zero]
 
 theorem neg_eq' : Neg.neg = (id : R → R) :=
   funext neg_eq
 
-@[scoped simp]
 theorem sub_eq_add (x y : R) : x - y = x + y := by rw [sub_eq_add_neg, neg_eq]
 
-theorem sub_eq_add' : HSub.hSub = ((· + ·) : R → R → R) :=
-  funext fun x => funext fun y => sub_eq_add x y
+/-- `-` and `+` are the same function in characteristic 2. -/
+@[scoped simp]
+theorem sub_eq_add' : HSub.hSub = (HAdd.hAdd : R → R → R) :=
+  funext₂ sub_eq_add
+
+theorem add_eq_iff_eq_add {a b c : R} : a + b = c ↔ a = c + b := by
+  rw [← sub_eq_iff_eq_add, sub_eq_add]
+
+theorem eq_add_iff_add_eq {a b c : R} : a = b + c ↔ a + c = b := by
+  rw [← eq_sub_iff_add_eq, sub_eq_add]
+
+@[scoped simp]
+protected theorem two_zsmul (x : R) : (2 : ℤ) • x = 0 := by
+  rw [two_zsmul, add_self_eq_zero]
```

Comment:
Perhaps also add a `two_smul` version for an arbitrary module, like [two_smul](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/Defs.html#two_smul)?

Reply:
Does that work? `two_smul` takes an argument `x : M` in the monoid, rather than `x : R` in the characteristic two semiring.

Reply:
I think that is a reasonable request but let's save it for another PR. @Command-Master feel free to PR such a change and ping me.